django/applications/catmaid/static/js/widgets/stack-viewer-grid.js

```diff
@@ -369,6 +369,54 @@
     scaleBarCbLabel.appendChild(scaleBarCb);
     this.controlsContainer.appendChild(scaleBarCbLabel);
 
+    var zoomInput = document.createElement('input');
+    zoomInput.setAttribute('type', 'text');
+    zoomInput.setAttribute('pattern', '^\-?\d?\.?\d*$');
+    zoomInput.setAttribute('size', '5');
+    // zoomInput.setAttribute('type', 'number');
+    // zoomInput.setAttribute('step', '0.1');
+    // zoomInput.setAttribute('max', '5');
+    // zoomInput.setAttribute('min', '-2');
```

Comment:
While the number type with spinner is nice in principal, I think the text type is fine due to not motivate performance killing actions. :-)

Alternatively, you could delay the actual change in zoom by maybe 200ms, so that individual changes on e.g. a spinner wouldn't trigger an immediate change, but only after a timeout (see `setTimeout()`). Anyhow, for now text is completely fine!